src/main/java/org/oscii/detokenize/Labeler.java

```diff
@@ -0,0 +1,74 @@
+package org.oscii.detokenize;
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.PeekingIterator;
+
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Infer label sequence from a preprocessor applied to a string.
+ * <p>
+ * TODO(denero) Infer capitalization and replacement
+ * <p>
+ * Note: The logic in this class could be removed if we exposed the CoreLabel
+ * sequence of the pre-processor, but since we may move away from CoreNLP, lets
+ * keep it for now.
+ */
+public class Labeler {
+    public List<TokenLabel> getLabels(String sentence, List<String> tokens) throws LabelException {
+        sentence = Normalizer.normalize(sentence, Normalizer.Form.NFC);
```

Comment:
I'll experiment with dropping this line, but it is applied during training only.